src/JuMPContainer.jl

```diff
@@ -99,6 +99,7 @@ end
 
 pushmeta!(x::JuMPContainer, sym::Symbol, val) = (x.meta[sym] = val)
 getmeta(x::JuMPContainer, sym::Symbol) = x.meta[sym]
+hasmeta(x::JuMPContainer, sym::Symbol) = haskey(x.meta, sym)
```

Comment:
Would be good to write this in such a way that it won't fail obscurely for user-defined types descended from ``JuMPContainer`` that don't have a ``meta`` field. Maybe something like
```jl
metadata(x::Union{JuMPArray,JuMPDict}) = x.meta
metadata(::JuMPContainer) = error("Useful error message here")
pushmeta!(x::JuMPContainer, sym::Symbol) = (metadata(x)[sym] = val)
getmeta(x::JuMPContainer, sym::Symbol) = metadata(x)[sym]
hasmeta(x::JuMPContainer, sym::Symbol) = haskey(metadata(x), sym)
```